libraries/AP_BLHeli/AP_BLHeli.cpp

```diff
@@ -1469,7 +1469,7 @@ void AP_BLHeli::read_telemetry_packet(void)
 
     uint8_t normalized_motor_idx = motor_idx - chan_offset;
 #if HAL_WITH_IO_MCU
-    if (AP_BoardConfig::io_dshot()) {
+    if (AP_BoardConfig::io_enabled()) {
```

Comment:
this seems quite convoluted, chan_offset is 8 if io_enabled(), zero otherwise, seems to be undoing what we have in init? can you explain what testing has been done on this, with and without IO enabled?

Reply:
can the subtraction motor_idx - chan_offset underflow?

Reply:
You are right, this can be cleaned up more. Since 4.5 the ESC index needs to reflect the servo index. This has been done in the bdshot code but was missed here.

Reply:
Fixed